tests/test_humidity_sensor.py

```diff
@@ -0,0 +1,44 @@
+from __future__ import absolute_import
+import unittest
+
+import mock
+
+from greenpithumb import humidity_sensor
+from greenpithumb import dht11_exceptions
+
+class HumiditySensorTest(unittest.TestCase):
+
+	def setUp(self):
+		self.mock_dht11 = mock.Mock()
+		self.humidity_sensor = humidity_sensor.HumiditySensor(
+			self.mock_dht11)
+
+	def test_humidity_50(self):
+		"""Value given should be value returned."""
+
+		self.mock_dht11.read().error_code = 0
+		self.mock_dht11.read().humidity = 50.0
```

Comment:
I'm actually kind of surprised this works. We might be relying on undefined behavior, meaning that it could change by chance in a minor update to Python or it might have different behavior on different systems.

What I'd expect to happen is that each `read()` creates a new, independent object that gets thrown away at the end of the line. But the `mock` library does a lot of crazy magic that I think makes this work for whatever reason right now, but we probably shouldn't depend on it.

I'd actually suggest fixing this in `setUp` so instead of keeping around `mock_dht11`, we keep around the result, so like:

```python
def setUp(self):
  self.mock_dht11_result = mock.Mock()
  mock_dht11 = mock.Mock(return_value=self.mock_dht11_result)
  self.humidity_sensor = humidity_sensor.HumiditySensor(mock_dht11)

  # The last two lines can actually reduce to:
  self.humidity_sensor = humidity_sensor.HumiditySensor(mock.Mock(return_value=self.mock_dht11_result))
```

Then in this test, we could do:

```python
def test_humidity_50(self):
  self.mock_dht11_result.error_code = 0
  self.mock_dht11_result.humidity = 50.0
  ...
```

Notice that we don't need to hang onto a reference to `mock_dht11` by assigning it to `self` because everything we need to manipulate can happen through `self.mock_dht11_result`.